server/app/lib/utils/cloud_forms/provider.rb

```diff
@@ -64,6 +64,25 @@ def self.add(cfme_ip, provider_params)
         }
         request_data = new_provider_form.request_data
         provider_added = agent.post("https://#{cfme_ip}/ems_infra/create/new?button=add", request_data, submit_headers)
+
+        show_list = agent.get("https://#{cfme_ip}/host/show_list", [], agent.page.uri)
+        while show_list.links_with(:href => %r'/host/show/1r', :text => "\n\t\t\t\t\n\t\t\t\t").count < provider_params[:hypervisors].count
```

Comment:
This is happening immediately after line 66
provider_added = agent.post("https://#{cfme_ip}/ems_infra/create/new?button=add", request_data, submit_headers)

which actually adds the RHEV provider. That's pretty instantaneous and triggers the refresh of the relationships and power state. It seems, that as long as all is good it takes about 35-40 seconds to create the hosts, and is what we're waiting on.

incremented counter sounds good along with the log message you mentioned. So I'll add those squash it and should be good to merge.

Thanks